### PR TITLE
feat(CI): Move CI to Github Actions

### PR DESCRIPTION
**Pull request by samoehlert**
Closes: #50 

Moving to Github Actions allows for better external collaboration as well as just a pure speed increase. This also means we can remove the dependency on Gitlab CI. We did remove a few of the built in processes from Gitlab CI file, but that's because we could turn on things in the settings directly (secret scanning and dependency scanning as well as Code Quality/SAST).

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -3,7 +3,8 @@ name: Build sphinx docs
 on:
   # Runs on pushes targeting the default branch
   push:
-    branches: ["main"]
```
Comment (grigorescu) [inline .github/workflows/docs.yml:6]: I think we want this to run always, but make the very last step (upload artifact) only run on main via this: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif

This way, we can ensure the Pages are getting built and at least not failing before it hits main.

File: .github/workflows/flake8.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Run flake8
+
+on:
+  # Runs on pushes targeting the default branch
```
Comment (grigorescu) [inline .github/workflows/flake8.yml:4]: I think this (and the next workflow) should always run; we can always explicitly skip it via https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/skipping-workflow-runs

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        env:
+          POSTGRES_USER: scram
+          POSTGRES_DB: test_scram
+        run: make coverage.xml
+
+      - name: Upload Coverage Report
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report
+          path: coverage.xml
```
Comment (grigorescu) [inline .github/workflows/pytest.yml:67]: Might want to consider https://github.com/marketplace/actions/cobertura-report

**Review by grigorescu — CHANGES_REQUESTED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (samoehlert) [reply .github/workflows/docs.yml:6]: Good point, I'll work on this.

**Review by grigorescu — DISMISSED** [review]

**Review by crankynetman — APPROVED** [review]
This looks great, nice work!

**Pull request merged**